### ISSUE TITLE
Multiple GPUs ： After assigning a GPU, all GPU will be trained 

### ISSUE THREAD

**Issue by codesun8** [ultralytics/yolov5#9880]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### Bug

shell:python3 -u  train_one_type.py --device 0,4 --batch-size 32 --cache --multi-scale --single-cls
![图片](https://user-images.githubusercontent.com/116338547/197133608-d7c5df1f-e3da-409f-80df-43faaa59c149.png)

### Environment

YOLO-YOLOV5 6.2 torch-1.12.0+cu116 CUDA0,1,2,3,4,5,6,7(NVIDIA A30 24GIB)
OS- centos7
python -python3.7.0

### Are you willing to submit a PR?

- [X] Yes I'd like to help by submitting a PR!

**Comment by glenn-jocher** [comment]
@sunyicode0012 I've seen this error occur in the past with certain commits and torch versions, but DDP works correctly with current master and latest torch. Can you verify that you are using the latest version of both?

**Comment by glenn-jocher** [comment]
Also I'd recommend Docker for all DDP trainings to minimize the risk of issues like this.

## Environments

YOLOv5 may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://bit.ly/yolov5-paperspace-notebook"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

**Comment by sunyicode0012** [comment]
@glenn-jocher Very glad you can reply to me. The version of the torch I use now is torch-1.12.0+cu116 and the version of yolo is V6.2. These two tools should all be the latest versions. There is a problem that my CUDA installation is 11.7. Is there any conflict? Is it related to the CUDA version?

**Comment by sunyicode0012** [comment]
@glenn-jocher  I think I know why.
[https://github.com/ultralytics/yolov5/issues/8395 ]() this issue tells me the answer
I use now is torch-1.12.0 
I should update to torch-1.12.1
Is that so?

**Comment by glenn-jocher** [comment]
@sunyicode0012 yes, that's correct! Upgrading to torch-1.12.1 should resolve the issue. Thank you for finding the related issue, and thank you for your patience.
